snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -90,5 +121,78 @@ def launch_instance(self) -> None:
     def setup_snapcraft(self) -> None:
         self.echoer.info('Setting up snapcraft in {!r}'.format(
             self.instance_name))
+        if common.is_snap():
+            self._inject_snapcraft()
+        else:
+            self._install_snapcraft()
+
+    def _install_snapcraft(self) -> None:
+        """Install snapcraft from the store."""
         install_cmd = ['sudo', 'snap', 'install', 'snapcraft', '--classic']
         self._run(install_cmd)
+
+    def _inject_snapcraft(self) -> None:
+        """Install snapcraft using local assets."""
+        # Make the snaps available to the provider
+        self._mount(mountpoint=self._SNAPS_MOUNTPOINT,
+                    dev_or_path=self._snaps_path_or_dev)
+        # TODO this should be properly fixed in snapd.
```

Comment:
Or using @mvo5 's hold trick

Reply:
Do we want to implement the hold trick in this PR?